cmd/track_bundle_test.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2022 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package cmd
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/empty"
+	"github.com/google/go-containerregistry/pkg/v1/mutate"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/google/go-containerregistry/pkg/v1/static"
+	"github.com/google/go-containerregistry/pkg/v1/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/tektoncd/pipeline/pkg/remote/oci"
+
+	"github.com/hacbs-contract/ec-cli/internal/image"
+)
+
+func Test_tektonBundleCollector(t *testing.T) {
+	defaultFetchImage := fetchImage
+	t.Cleanup(func() {
+		fetchImage = defaultFetchImage
+	})
+
+	cases := []struct {
+		name     string
+		kinds    []string
+		err      error
+		expected []string
+	}{
+		{
+			name: "failed to fetch image",
+			err:  errors.New("expected"),
+		},
+		{
+			name:     "a task and a pipeline",
+			kinds:    []string{"task", "pipeline"},
+			expected: []string{"task", "pipeline"},
+		},
+		{
+			name:     "multiple tasks and a pipelines",
+			kinds:    []string{"task", "pipeline", "pipeline", "task", "task"},
+			expected: []string{"task", "pipeline"},
+		},
+	}
```

Comment:
Do we need to add coverage for just `task` and just `pipeline`? Feels odd to not have the main use case.

Reply:
There is [one already](https://github.com/hacbs-contract/ec-cli/pull/109/files#diff-d3aa3c8fc83384b5f3aee62abaafbda79e331eb4e648bdcf66f242eb1c29de7bR54)

Reply:
I meant:
```go
		{
			name:     "tasks only",
			kinds:    []string{"task", "task", "task"},
			expected: []string{"task"},
		},
		{
			name:     "pipelines only",
			kinds:    []string{"pipeline", "pipeline", "pipeline"},
			expected: []string{"pipeline"},
		},
```

Probably overkill...